src/server.js

```diff
@@ -44,7 +45,39 @@ export class Server {
                 if (response instanceof Error) {
                     return Promise.reject(response);
                 } else {
-                    return Promise.reject(response.data);
+                    if (response.data.submission_result) {
+                        let transactionResult = xdr.TransactionResult.fromXDR(new Buffer(response.data.submission_result, 'hex'));
+                        // Is there a better way to do it?
```

Comment:
The TransactionResult contains more information than just the top level failure, for example it contains results for each operation. Someone using this api to submit multiple operations with multiple source accounts in one transaction may be interested in operation level results - for instance, pulling the problem operation out of the built transaction and resubmitting the transaction. 

Right now, you simply throw the top level transaction error so the consumer of the API loses the entire TransactionResult. I agree that parsing the TransactionResult should be done at this level, but the consumer should also have access to the TransactionResult when the promise is rejected.